hazelcast/src/main/java/com/hazelcast/sql/impl/expression/Expression.java

```diff
@@ -70,4 +70,6 @@ default T eval(Row row, ExpressionEvalContext context, boolean useLazyDeserializ
      * @return the return query data type of this expression.
      */
     QueryDataType getType();
+
+    boolean isCooperative();
```

Comment:
I think that `isCooperative` should be checked in more places, and maybe currently we could just reject non-cooperative expressions in places which to do not implement them. Example: `remainingFilter` for `MapIndexScanP` and other places which do filtering and projections inline, not as separate `Calc` operator

Reply:
Except for `GET_DDL`, I don't know any non-cooperative expression. `GET_DDL` itself is a service function and (by idea) shouldn't take part in any high-performing computations.

Reply:
`GET_DDL` can be used anywhere, where expression is expected. If processor evaluating expression does not support non-cooperative expressions it should reject them.

Reply:
We now have a check in every non-cooperative expression that will raise an exception if it's running on a cooperative thread. So if we missed the cooperativity check for some expression evaluation, it will fail fast.